packages/components/src/duotone-picker/utils.ts

```diff
@@ -38,20 +45,26 @@ export function getDefaultColors( palette ) {
 					current.brightness >= max.brightness ? current : max,
 				];
 			},
-			[ { brightness: 1 }, { brightness: 0 } ]
+			[
+				{ brightness: 1, color: '' },
+				{ brightness: 0, color: '' },
+			]
 		)
 		.map( ( { color } ) => color );
 }
```

Comment:
An interesting example of how some clever functional code may not play nice with TS 😆 The initial values (`[ { brightness: 1 }, { brightness: 0 } ]`) will never be remaining at the point of the final `.map()`, but TS cannot know that.

I _think_ this is probably the least annoying/invasive way around it.

Reply:
> I think this is probably the least annoying/invasive way around it.

Agreed!

Although annoying at times, I'm glad that TS is able to pick up these edge cases. The fact that the author of these lines of code was clever in writing the original implementation doesn't necessarily mean that a malformed color palette (e.g with out of scale brightness values) or another developer making amends to the algorithm could introduce a bug later in time. TS checks are great in avoiding such events

Reply:
Though for this particular function I would rely on some unit tests more than TS 🫣

Reply:
That's true! I guess in my mind I implicitly considered TS static linting as a the first test that runs against the code.